src/creatures/players/player.cpp

```diff
@@ -5428,12 +5434,12 @@ uint16_t Player::getFreeBackpackSlots() const
 	return counter;
 }
 
-void Player::addItemImbuementStats(const Imbuement* imbuement)
+void Player::addItemImbuementStats(const Imbuement* imbuement, const ItemType& it)
 {
 	bool requestUpdate = false;
 	// Check imbuement skills
 	for (int32_t i = SKILL_FIRST; i <= SKILL_LAST; ++i) {
-		if (imbuement->skills[i]) {
+		if (imbuement->skills[i] && (i != SKILL_CRITICAL_HIT_CHANCE || !it.abilities || !it.abilities->skills[i])) {
```

Comment:
this is a bit too hacky, would love to see an explanation of the need to come up with a better solution or at least a nicer if statement.